### ISSUE TITLE
Recycle bin feature

### ISSUE THREAD

**Issue by thesushil** [ildar-shaimordanov/psubst#18]
Please add a feature so that the files deleted from the virtual drive go into recycle bin. Thank you.

**Comment by ildar-shaimordanov** [comment]
It is hardly possible to implement the requested feature in easy way so to make it working automatically. I recommend you to do it manually following the steps as it's explained in this QA: https://superuser.com/questions/222747/recycle-bin-on-substed-drives.

**Comment by Artoria2e5** [comment]
Please reopen this. Your objection is on the ground that it "only works for a single subst", while in reality this is only causes by a fixed GUID. A GUID can be trivially generated using PowerShell. If we don't have PS, we can even just use 123e4567-e89b-42d3-a456-1337095b57XY for all we care, where XY is the hexadecimal ASCII of the drive number.

**Comment by ildar-shaimordanov** [comment]
There are two issues already about the same thing: the issue #19 and this one.

Besides that there is stuff mentioned in https://github.com/ildar-shaimordanov/psubst/blob/master/TODO. I think the recycler is the tool you could like as the workaround to enable recycle bin for substed drives.

**Comment by Artoria2e5** [comment]
The UUID thing I have solved with a new subroutine
```bat
:psubst_uuid
echo WScript.Echo Hex(Asc(WScript.Arguments(0))) > "%TEMP%\%~n0.hex.vbs"
for /f %%L in ('"%TEMP%\%~n0.hex.vbs" "!psubst_persist_disk!"') do set "uuid={123e4567-e89b-42d3-a456-1337950b57%%L}"
goto :EOF
```

But the elevated reg thing is starting to look awful as I must run multiple reg commands, so multiple sudo dances. Duh...

**Comment by ildar-shaimordanov** [comment]
I see necessity in implementing this feature. However let's minimize any pollution to the main script. Enabling the recycle bin is not the main functionality of the psubst tool. It should be implemented as another tool turning on/off the recycle bin for SUBSTed drives.

**Comment by sergeevabc** [comment]
@ildar-shaimordanov, I disagree. Using SUBST is like plugging in another internal or external drive and Recycle bin is a property of any _fixed_ drive that an average user _expects_ to have out of the box. E.g. if C:\ is SUBST’ed as X: and C: uses Recycle bin, then K: is expected to inherit that. It is an enlightened geeky admin who is aware of the current pitfalls, but an average user suffers. To reduce the amount of suffering, this solution should be improved. don’t know how yet, since am not a developer, but you’re.
